### ISSUE TITLE
No suggestions/completion for vars in typed subclass

### ISSUE THREAD

**Issue by jozefchutka** [HaxeFoundation/intellij-haxe#366]
Working with following simplified class architecture, I can get **a** and **b** params to auto suggest ".z" property, but **c** would not provide any suggestion (ctrl+space = "No suggestions"). This seems to me that suggestions for typed classes do not work further down then passing type more then 1 level of class.

``` haxe
class A extends B<Z> {
    var a:Z;
    function new()
    {
        a.z;
        b.z;
        c.
    }
}

class B<T> extends C<T> {
    var b:T;
}

class C<T> {
    var c:T;
}

class Z {
    public var z:Int;
}
```

Working with IntelliJ Idea 14.1.5, haxe support 0.9.6, haxe 3.2.0

**Comment by EBatTiVo** [comment]
This is a known issue with template (generic) types.  The problem lies in the algorithm that walks the type tree: The subtype is noted as a T, but there is a disconnect between the actual type and it's parameterized value (meaning T is known, but it doesn't know that T is really a Z).  I remember working on this about six months ago and I'm not sure where I left it.

**Comment by EricBishton** [comment]
Fixed in 0.11.0-RC3.
